File: app/src/main/java/com/semisonfire/cloudgallery/ui/dialogs/BottomDialogFragment.java
package com.semisonfire.cloudgallery.ui.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetBehavior;
import android.support.design.widget.BottomSheetDialog;
import android.view.View;
import android.view.ViewGroup;

import com.semisonfire.cloudgallery.R;
import com.semisonfire.cloudgallery.ui.dialogs.base.BaseDialogFragment;

public class BottomDialogFragment extends BaseDialogFragment {

    private BottomSheetDialog dialog;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        dialog = new BottomSheetDialog(context);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        View sheetView = View.inflate(getContext(), R.layout.dialog_bottom, null);
        bind(sheetView);

        dialog.setContentView(sheetView);

        BottomSheetBehavior behavior = BottomSheetBehavior.from((View) sheetView.getParent());
        dialog.setOnShowListener(dialogInterface -> behavior.setPeekHeight(sheetView.getHeight()));

        return dialog;
    }

    private void bind(View sheetView) {
        View.OnClickListener clickListener = view -> {
            if (getDialogListener() != null) getDialogListener().onItemClick(this.getDialog(), view);
        };

        ViewGroup mCamera = sheetView.findViewById(R.id.container_camera);
        mCamera.setOnClickListener(clickListener);

        ViewGroup mGallery = sheetView.findViewById(R.id.container_gallery);
        mGallery.setOnClickListener(clickListener);
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/dialogs/BottomDialogFragment.kt
package com.semisonfire.cloudgallery.ui.dialogs

import android.app.Dialog
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.semisonfire.cloudgallery.R

class BottomDialogFragment : DialogFragment() {

    var dialogListener: DialogListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val context = requireContext()

        val bottomSheetDialog = BottomSheetDialog(context)
        val sheetView = View.inflate(context, R.layout.fragment_upload_chooser, null)
        bind(sheetView)

        bottomSheetDialog.setContentView(sheetView)

        val behavior = BottomSheetBehavior.from(sheetView.parent as View)
        return bottomSheetDialog.apply {
            setOnShowListener {
                behavior.setPeekHeight(sheetView.height)
            }
        }
    }

    private fun bind(sheetView: View) {
        val clickListener = View.OnClickListener {
            dialogListener?.onItemClick(it)
            dismiss()
        }

        val camera = sheetView.findViewById<TextView>(R.id.container_camera)
        camera.setOnClickListener(clickListener)

        val gallery = sheetView.findViewById<TextView>(R.id.container_gallery)
        gallery.setOnClickListener(clickListener)
    }
}